SYCL/Basic/kernel_info.cpp

```diff
@@ -51,4 +51,12 @@ int main() {
   const size_t prefWGSizeMult = krn.get_info<
       info::kernel_device_specific::preferred_work_group_size_multiple>(dev);
   assert(prefWGSizeMult > 0);
+
+  assert(dev.get_info<sycl::info::device::device_type>() !=
+         sycl::info::device_type::custom);
+  try {
+    krn.get_info<sycl::info::kernel_device_specific::global_work_size>(dev);
```

Comment:
I don't like us asserting that it is not a custom device, as if we suddenly have a custom device this test will just fail in a case where that device would be picked up. Instead it would make sense to check that if it doesn't throw it is on a custom device:
```suggestion
  try {
    krn.get_info<sycl::info::kernel_device_specific::global_work_size>(dev);
    assert(dev.get_info<sycl::info::device::device_type>() ==
           sycl::info::device_type::custom);
```